package/samplers/carbo/sampler.py

```diff
@@ -244,6 +246,18 @@ def infer_relative_search_space(
 
         return search_space
 
+    def get_best_trial(self, study: Study) -> FrozenTrial:
+        complete_trials = study._get_trials(
+            deepcopy=False, states=(TrialState.COMPLETE,), use_cache=False
+        )
+        if len(complete_trials) == 0:
+            raise ValueError("No complete trials found in the study.")
+
+        best_idx = np.argmax(
+            [t.system_attrs.get(_BEST_ACQF_KEY, -np.inf) for t in complete_trials]
+        )
```

Comment:
The definition of the acquisition function (equation (12)) uses an **optimistic** evaluation, whereas equation (14) uses a **pessimistic** evaluation, so the value of the acquisition function cannot be used here.

Reply:
`best_acqf_val` here is indeed identical to Eq. (14).
The acquisition function equivalent to Eq. (12) is defined [here](https://github.com/optuna/optunahub-registry/pull/277/files#diff-de75d2c76231aeedb7b7e3e34d91f9301ebcb7ce0b44d8f7a2724f26f445ea7dR86-R92).

Reply:
Oop! Sorry, I misunderstood the implementation of the `suggest_by_carbo` function.